dropwizard/src/main/java/com/linecorp/armeria/server/dropwizard/ArmeriaServerFactory.java

```diff
@@ -0,0 +1,222 @@
+    }
+
+    private void setupArmeria(final Server server, final MetricRegistry metricRegistry) {
+        LOGGER.debug("Building Armeria Server");
+        final JettyService jettyService = JettyService.forServer(server);
+        this.serverBuilder = com.linecorp.armeria.server.Server.builder()
+                .meterRegistry(DropwizardMeterRegistries.newRegistry(metricRegistry))
+                .serviceUnder("/", jettyService);
+
+        if (this.connector != null) {
+            try {
+                setupAmeriaHttp(this.serverBuilder, this.connector);
+            } catch (SSLException | CertificateException e) {
+                LOGGER.error("Unable to define TLS Server", e);
+            }
+        }
+
+        // TODO: Add more items to server builder via Configuration
+        // https://line.github.io/armeria/advanced-production-checklist.html
+        serverBuilder.maxNumConnections(getMaxQueuedRequests());
+        serverBuilder.blockingTaskExecutor(Executors.newFixedThreadPool(getMaxThreads()), true);
+        serverBuilder.maxRequestLength(this.maxRequestLength.toBytes());
+        serverBuilder.idleTimeoutMillis(getIdleThreadTimeout().toMilliseconds());
+        serverBuilder.gracefulShutdownTimeout(
+                Duration.ofSeconds(30L),
+                Duration.ofMillis(getShutdownGracePeriod().toMilliseconds()));
+    }
+
+    private void setupAmeriaHttp(final ServerBuilder sb, final ConnectorFactory connector)
+        throws SSLException, CertificateException {
+        if (connector instanceof ArmeriaHttpConnectorFactory) {
+            LOGGER.debug("Building Armeria HTTP Server");
+            final ArmeriaHttpConnectorFactory factory = (ArmeriaHttpConnectorFactory) connector;
+            sb.http(factory.getPort());
+            // TODO: More HTTP1 settings
+            sb.http1MaxHeaderSize((int) factory.getMaxRequestHeaderSize().toBytes());
+        } else if (connector instanceof ArmeriaHttpsConnectorFactory) {
+            LOGGER.debug("Building Armeria HTTPS Server");
+            final ArmeriaHttpsConnectorFactory factory = (ArmeriaHttpsConnectorFactory) connector;
+            sb.https(factory.getPort());
+            if (factory.isSelfSigned()) {
+                sb.tlsSelfSigned();
+            }
+            final File cert = new File(factory.getKeyCertChainFile());
+            final File keyStore = new File(factory.getKeyStorePath());
+            if (factory.isValidKeyStorePassword()) {
+                sb.tls(cert, keyStore, factory.getKeyStorePassword());
+            } else {
+                sb.tls(cert, keyStore);
+            }
+            // TODO: More HTTPS settings
+        } else {
+            // TODO: HTTP isn't required for just gRPC servers. Is a null connector representative of that?
+            throw new ClassCastException("server.connector must be one of types " +
+                  ArmeriaHttpConnectorFactory.TYPE + " or " + ArmeriaHttpsConnectorFactory.TYPE);
```

Comment:
`connector` has a default value of `ArmeriaHttpConnectorFactory`... 

TODO: Try removing the default and using gRPC / Thrift only